packages/dai-plugin-mcd/test/SavingsService.spec.js

```diff
@@ -15,24 +22,29 @@ describe('Savings Service', () => {
     maker = await mcdMaker();
     service = maker.service(ServiceRoles.SAVINGS);
     dai = maker.getToken(MDAI);
-
     proxyAddress = await maker.service('proxy').ensureProxy();
+    nonceService = maker.service('nonce');
+    currentAddress = maker.service('web3').currentAddress();
     await dai.approveUnlimited(proxyAddress);
+    transactionCount = nonceService._counts[currentAddress];
   });
 
   afterAll(async () => {
     await maker.service('allowance').removeAllowance('MDAI', proxyAddress);
   });
 
   beforeEach(async () => {
-    const amountRemaining = await service.balance();
-    if (amountRemaining.gt(0)) await service.exit(MDAI(amountRemaining));
+    snapshotId = await takeSnapshot();
+  });
+
+  afterEach(async () => {
+    await restoreSnapshot(snapshotId);
+    nonceService._counts[currentAddress] = transactionCount;
```

Comment:
since you're doing this bit the same way three times, you could move it into the helper functions, e.g.:
```js
beforeEach(() => {
  snapshotData = await takeSnapshot(maker); 
  // object containing snapshot id and transaction count
  // you could omit the maker arg and then just get an object with snapshot id
})

afterEach(() => {
  await restoreSnapshot(snapshotData);
})